### PR TITLE
implemented rateLimit for status change

### PR DESCRIPTION
**Pull request by AdrienRem**

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/UserInfectionTest.java [hunk 1]
```diff
@@ -79,9 +80,12 @@ public boolean updateStatus(Carrier.InfectionStatus stat) {
         };
         fragment.getLocationService().setAnalyst(analyst);
         receiver = fragment.getLocationService().getReceiver();
+        fragment.getActivity().getSharedPreferences("UserInfectionPrefFile", Context.MODE_PRIVATE)
```
Comment (TheBlueHawk) [inline app/src/androidTest/java/ch/epfl/sdp/UserInfectionTest.java:83]: Could you please me explain what is "Context.MODE_PRIVATE"?

File: app/src/main/java/ch/epfl/sdp/fragment/UserInfectionFragment.java [hunk 2]
```diff
@@ -166,24 +175,41 @@ private void getLoggedInUser() {
...
+                sendRecoveryToFirebase();
+                setInfectionColorAndMessage(false);
+                modifyUserInfectionStatus(userName, false,
+                        value -> {
+                            //infectionUploadView.setText(String.format("%s at %s", value, Calendar.getInstance().getTime()))
+                        });
+            }
+        }else {
+            Toast.makeText(getActivity().getApplicationContext(),
+                    "Your health seems to be changing fast, Ignoring", Toast.LENGTH_LONG).show();
```
Comment (TheBlueHawk) [inline app/src/main/java/ch/epfl/sdp/fragment/UserInfectionFragment.java]: Hardcoded string, should be put in string resources both for English and French

**Review by TheBlueHawk — CHANGES_REQUESTED** [review]
All good except for the hardcoded string. It shouldn't take long to fix!

File: app/src/androidTest/java/ch/epfl/sdp/UserInfectionTest.java (hunk 1 above)
Comment (AdrienRem) [reply app/src/androidTest/java/ch/epfl/sdp/UserInfectionTest.java:83]: It just means that the preference object is not shared with other applications. So it is stored inside the app private folder

File: app/src/main/java/ch/epfl/sdp/fragment/UserInfectionFragment.java [hunk 3]
```diff
@@ -166,24 +175,41 @@ private void getLoggedInUser() {
...
-            service.getAnalyst().updateStatus(Carrier.InfectionStatus.HEALTHY);
-            sendRecoveryToFirebase();
-            setInfectionColorAndMessage(false);
-            modifyUserInfectionStatus(userName, false,
-                    value -> {
-                        //infectionUploadView.setText(String.format("%s at %s", value, Calendar.getInstance().getTime()))
-                    });
+
+        Date currentTime = Calendar.getInstance().getTime();
+        /* get 1 jan 1970 by default. It's definitely wrong but works as we want t check that
```
Comment (TheCrazyKing) [inline app/src/main/java/ch/epfl/sdp/fragment/UserInfectionFragment.java:180]: Why is it wrong ? every time that we have on our app is relative to that famous 1 jan 1970.

**Review by TheCrazyKing — CHANGES_REQUESTED** [review]
I see no tests that check nothing is made if you change your status too often. You should definitely add one :)

File: app/src/main/java/ch/epfl/sdp/fragment/UserInfectionFragment.java (hunk 2 above)
Comment (AdrienRem) [reply app/src/main/java/ch/epfl/sdp/fragment/UserInfectionFragment.java]: I acted on your recommendations Beloved and Respected Leader, please see commit ee0b1a1

**Comment by AdrienRem** [comment]
> I see no tests that check nothing is made if you change your status too often. You should definitely add one :)

I totally did not forget to add a test ;) Please see commit df6e34b

File: app/src/main/java/ch/epfl/sdp/fragment/UserInfectionFragment.java (hunk 3 above)
Comment (AdrienRem) [reply app/src/main/java/ch/epfl/sdp/fragment/UserInfectionFragment.java:180]: It's the epoch, the beginning of time for unix people. It's akin to the birth of Jesus for Christians.

**Review by TheBlueHawk — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/fragment/UserInfectionFragment.java (hunk 3 above)
Comment (TheCrazyKing) [reply app/src/main/java/ch/epfl/sdp/fragment/UserInfectionFragment.java:180]: Yes ahah. But my quesiton swas about your comment : " It's definitely wrong " - nevermind.

**Review by TheCrazyKing — APPROVED** [review]
With tests it is all fine nice job !

**Pull request merged**